client/js/app.js

```diff
@@ -0,0 +1,56 @@
+const btn = document.querySelector('#post-form');
+const main = document.querySelector('main');
+
+function submitPost(e){
+    e.preventDefault();
+    const postData = {
+        title: e.target.title.value,
+        name: e.target.name.value,
+        body: e.target.body.value
+    }
+
+    const options = {
+        method: 'POST',
+        body: JSON.stringify(postData),
+        headers: {
+            "Content-Type": "application/json"
+        }
+    };
+
+    fetch('http://localhost:3000/posts/', options)
+        .then(d=>d.json())
+        .then(d => window.location.hash = `/posts/${d.id}`)
+        .catch(console.warn)
+}
+
+btn.addEventListener('submit', submitPost);
+window.addEventListener('hashchange', updateContent)
+
+function updateContent(){
+    let hash = window.location.hash.substring(8);
+    updateMain(hash);
```

Comment:
As update main is only called for this context, no need to split into two functions, could have just put the code of `updateMain` following the `let hash` line

Reply:
im a modularity addict